File: src/net/zomis/tttultimate/HasSub.java
package net.zomis.tttultimate;

public interface HasSub<T> extends Winnable {
	T getSub(int x, int y);
	Iterable<TTWinCondition> getWinConds();
	int getSizeX();
	int getSizeY();
	int getConsecutiveRequired();
	boolean hasSubs();
}

File: uttt-common/src/main/kotlin/net/zomis/tttultimate/HasSub.kt
package net.zomis.tttultimate

/**
 * Interface for classes that can contain other objects, 'subs', in a rectangular way
 *
 * @param <T> Type of sub
</T> */
interface HasSub<out T> : Winnable {
    val winConds: Iterable<TTWinCondition>
    val sizeX: Int
    val sizeY: Int
    val consecutiveRequired: Int
    fun getSub(x: Int, y: Int): T?
    fun hasSubs(): Boolean
}
